### ISSUE TITLE
sig-windows flaky tests

### ISSUE THREAD

**Issue by jsturtevant** [kubernetes/kubernetes#92771]
**Which jobs are flaking**:
aks-engine-azure-windows-1903-master
aks-engine-azure-windows-1909-master
aks-engine-azure-windows-master-staging

**Which test(s) are flaking**:
the whole test suite times out

**Testgrid link**:
https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-1903-master
https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-1909-master
https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-master-staging

**Reason for failure**:
These tests use aks-engine to do the provisioning and run a custom script extension to taint the control nodes so the windows containers tests run only on the windows nodes.  The linux machine is rebooting due to apt package updates during the script running [causing it to hang](https://github.com/Azure/custom-script-extension-linux/issues/106).  This doesn't happen every time.

**Anything else we need to know**:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

/sig windows
/kind flake

**Comment by jsturtevant** [comment]
@chewong for tracking the changes

/assign

**Comment by jsturtevant** [comment]
the output from the `artifacts/master_provisioning/handler.log`:

```
time=2020-07-02T17:48:30Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 file=0 event="download start"
time=2020-07-02T17:48:31Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 file=0 event="download complete" output=/var/lib/waagent/custom-script/download/1
time=2020-07-02T17:48:31Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 event="executing command" output=/var/lib/waagent/custom-script/download/1
time=2020-07-02T17:48:31Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 event="executing protected commandToExecute" output=/var/lib/waagent/custom-script/download/1
Not writing a placeholder status file, already exists: /var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.1.3/status/1.status
+ nohup /var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.1.3/bin/custom-script-extension enable
time=2020-07-02T17:49:45Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 event=start
time=2020-07-02T17:49:45Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 event=pre-check
time=2020-07-02T17:49:45Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 event="comparing seqnum" path=mrseq
time=2020-07-02T17:49:45Z version=v2.1.3/git@4cd2b9f-clean operation=enable seq=1 event=exit message="the script configuration has already been processed, will not run again"
```

the restart from `artifacts/master_provisioning/cluster-provision.log`:

```
+ mkdir -p /opt/azure/containers
+ touch /opt/azure/containers/provision.complete
Shutdown scheduled for Thu 2020-07-02 17:49:06 UTC, use 'shutdown -c' to cancel.
+ exit 0
+ exitAndFlush
```

**Comment by jsturtevant** [comment]
Quick update we made a few changes to the [pre-provisioning script](https://github.com/kubernetes-sigs/windows-testing/pull/180) and the way [aks-engine loads the binaries](https://github.com/Azure/aks-engine/pull/3574) that made this should make this more stable.  

Still waiting for a good signal due to issues with  #80917 and #90948 which causes all tests to fail and was resolved in #93263

**Comment by jsturtevant** [comment]
The cluster provisioning is no longer running into this issue.  

/close
